### PR TITLE
Propagate esbuild `no-remote-exec` from tags

### PR DESCRIPTION
**Pull request by siggisim**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

I've tested this change with a local patch, and it works great with or without RBE enabled. Happy to add tests if it makes sense for a change like this.

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently esbuild `execution_requirements` are hardcoded to disable remote execution. When running a build with remote execution enabled, the esbuild action will always run locally.

If running with `--spawn_strategy=remote`, you get the following error:
```
ERROR: /Users/siggi/Code/buildbuddy/app/BUILD.bazel:13:8: Bundling Javascript app/app.mjs [esbuild] failed: No usable spawn strategy found for spawn with mnemonic Action.  Your --spawn_strategy, --genrule_strategy and/or --strategy flags are probably too strict. Visit https://github.com/bazelbuild/bazel/issues/7480 for migration advice
```

Issue Number: N/A

## What is the new behavior?
If the `no-remote-exec` tag is set on the esbuild rule, that value is propagated down to the action's `execution_requirements`.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/esbuild/esbuild.bzl [hunk 1]
```diff
@@ -115,15 +115,17 @@ def _esbuild_impl(ctx):
     if ctx.attr.max_threads > 0:
         env["GOMAXPROCS"] = str(ctx.attr.max_threads)
 
+    execution_requirements = {}
+    if "no-remote-exec" in ctx.attr.tags:
```
Comment (alexeagle) [inline packages/esbuild/esbuild.bzl:119]: is this a breaking change, in that users will have to add this tag after the next release?

@mattem I'm not clear why no-remote-exec was required here, do you know?

Comment (siggisim) [reply packages/esbuild/esbuild.bzl:119]: Only if those users have RBE enabled, and don't want this particular action to happen on RBE for whatever reason. While the behavior may change (for those users), it shouldn't really break anyone's build.

Discussed this a little on Friday with @mattem on Slack, he wasn't sure why that execution_requirements was set initially:
https://bazelbuild.slack.com/archives/CEZUUKQ6P/p1616796872102700

I did test this change on RBE and the rule works just fine remotely.

Happy to do this in a different way if you prefer and make disabling RBE the default - just seems like the tag approach is somewhat standard.

Comment (mattem) [reply packages/esbuild/esbuild.bzl:119]: I had a think about this, and I think it's just some copypasta from when I copied it from evertz internal.
If this is working as expected on RBE then it should be fine, but there is the point about breaking change, albeit small imo.

Comment (siggisim) [reply packages/esbuild/esbuild.bzl:119]: Let me know what you think the best way to proceed is - happy to make any changes or mark the PR as introducing a breaking change if you prefer.

Would love to get this landed, but can always stick with the local fork approach until we're ready!

Comment (alexeagle) [reply packages/esbuild/esbuild.bzl:119]: If we were really worried about the breaking change, we'd make your new attribute default match the current behavior. But I don't think we are, plus the esbuild package is quite new so I'm less concerned about dormant code that depends on it.

**Review by alexeagle — APPROVED** [review]

**Pull request merged**